driver/driver.go

```diff
@@ -89,11 +95,26 @@ type Driver interface {
 	Parse(ctx context.Context, src string, opts *ParseOptions) (nodes.Node, error)
 }
 
+// Version information for driver or the server.
+type Version struct {
+	Version string
+	Build   *time.Time
+}
+
+// Host is an extended Driver interface that also includes the list of supported
+// languages as well as the application version.
+type Host interface {
+	Driver
+	// Version returns a version of this driver or a version of the server that hosts it.
+	Version(ctx context.Context) (Version, error)
+	// Languages returns a list of manifests for languages supported by this driver or the server.
+	Languages(ctx context.Context) ([]manifest.Manifest, error)
```

Comment:
Having read further: We have several places where we do shenanigans to dereference pointers because of this. We could simplify those too.

Reply:
I'm not sure this will change anything. Most places that you mentioned convert the value to a different type so we will have to make a new slice regardless of the pointer in this one.

You are right that the structure is pretty large, but at the same time users will usually iterate over it, so it should be faster with the slice of values since it's a reference.

Also, it's considerably rare for a Go code to return slices with pointers unless the uniqueness of the pointer matters. I may be wrong, but this is what I've noticed in most of the libraries in the wild. I assumed the general rule is more in line of:
- Use the pointer if the value is unique or is mutable.
- If the method returns a value of the type, use pointer if it's large and value if it's small.
- For slices, use the value since the slice itself is a reference anyway.

Reply:
Pointer slices are pretty common in generated code—the Go protobuf plugin, for example, uses them extensively (basically, any repeated message field). Of course, for primitive types or small structures it doesn't really make sense to do it, which is why I raised the question.

Here, though, it seems clear we are pulling in both directions: We have a large struct that we handle by its pointer in most places, except here. I think it would make sense to try and be consistent about it—either returning it by value everywhere, and keeping values in the slice, or using a slice of pointers. 

My own experience has been using pointer slices (maybe mostly because of heavy protobuf use), but I'm mainly interested in making the API a bit more predictable.